ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStreamWriter.java

```diff
@@ -41,62 +40,134 @@
 import java.io.EOFException;
 import java.io.IOException;
 import java.io.OutputStream;
+import java.io.StringWriter;
 import java.nio.charset.StandardCharsets;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
 import java.util.List;
+import java.util.Locale;
 import java.util.Map;
 import java.util.stream.Collectors;
 
+import static io.confluent.ksql.rest.server.resources.streaming.TopicStreamWriter.Format.getFormatter;
+
 
 public class TopicStreamWriter implements StreamingOutput {
+  public enum Format {
+
+    UNDEFINED {
+  },
+    AVRO {
+      private String topicName;
+      private KafkaAvroDeserializer avroDeserializer;
+
+      @Override
+      public boolean isFormat(String topicName, ConsumerRecord<String, Bytes> record,
+          SchemaRegistryClient schemaRegistryClient) {
+        this.topicName = topicName;
+        try {
+          avroDeserializer = new KafkaAvroDeserializer(schemaRegistryClient);
+          avroDeserializer.deserialize(topicName, record.value().get());
```

Comment:
Is `isFormat` always called before `print`. If so, given how expensive avro deserialization is, would it make sense to cache this and return the cached value in the `print` method?

Reply:
No - its only called on the first message received and when the Format == UNDEFINED